kview/src/main/java/dev/ikm/komet/kview/klfields/KlFieldHelper.java

```diff
@@ -108,17 +116,62 @@ public static List<ObservableField<?>> generateObservableFieldsAndNodes(ViewProp
-
             items.add(node);
         };
         generateSemanticUIFields(viewProperties, semanticEntityVersionLatest, generateConsumer);
 
         return observableFields;
     }
 
+    public static int generateHashValue(Latest<SemanticEntityVersion> semanticEntityVersionLatest, ViewProperties viewProperties ) {
+        List<ObservableField<?>> observableFieldsList = new ArrayList<>();
+        Consumer<FieldRecord<Object>> fieldRecordConsumer = (fieldRecord) -> {
+            ObservableField<?> writeObservableField = obtainObservableField(viewProperties, semanticEntityVersionLatest, fieldRecord, false);
+            ObservableField<?> observableField = new ObservableField<>(writeObservableField.field(), false);
+            observableFieldsList.add(observableField);
+        };
+        generateSemanticUIFields(viewProperties, semanticEntityVersionLatest, fieldRecordConsumer);
+        return calculteHashValue(observableFieldsList);
+    }
+
+    /**
+     * This method will return the latest commited version.
+     * //TODO this method can be generalized to return latest<EntityVersion> As of now it is just returning SemanticEntityVersion.
+     * // TODO need to implement logic for create Semantic.
+     *
+     * @return entityVersionLatest
+     * */
+    public static Latest<SemanticEntityVersion> retrieveCommittedLatestVersion(EntityFacade entityFacade, ViewProperties viewProperties) {
+        AtomicReference<Latest<SemanticEntityVersion>> entityVersionLatest = new AtomicReference<>();
+
+        StampCalculator stampCalculator = viewProperties.calculator().stampCalculator();
+        //retrieve latest semanticVersion
+        Latest<SemanticEntityVersion> semanticEntityVersionLatest = stampCalculator.latest(entityFacade.nid());
+        ObservableSemantic observableSemantic = ObservableEntity.get(semanticEntityVersionLatest.get().nid());
+        ObservableSemanticSnapshot observableSemanticSnapshot = observableSemantic.getSnapshot(viewProperties.calculator());
+        //Get list of previously committed data sorted in latest at the top.
+        ImmutableList<ObservableSemanticVersion> observableSemanticVersionImmutableList = observableSemanticSnapshot.getHistoricVersions();
+        // Filter out Uncommitted data. Data whose time stamp parameter is Long.MAX_VALUE. and get the 1st available.
+        Optional<ObservableSemanticVersion> observableSemanticVersionOptional = observableSemanticVersionImmutableList.stream().filter(p -> p.stamp().time() != Long.MAX_VALUE).findFirst();
+
+        observableSemanticVersionOptional.ifPresentOrElse( (p) -> {
+            entityVersionLatest.set(new Latest<>(p));
+        }, () -> {entityVersionLatest.set(semanticEntityVersionLatest);});
+
+
+        return entityVersionLatest.get();
+    }
+
+    public static int calculteHashValue(List<ObservableField<?>> observableFieldsList ) {
```

Comment:
Can we also include the Semantic's reference component's `publicId`?
My reasoning: During create (new Semantic) the user can change the reference component. the hash is stating any change. By default a reference component during created would be `TinkarTerms.ANONOUMOUS_CONCEPT` (I can't remember).